### DISCUSSION TITLE
Cannot install IndyProtocols70.bpl - @Idglobal@UTCTimeToLocalTime$qqrx16System@Tdatetime could not located in the dynamic link library

### DISCUSSION THREAD

**Discussion by khwong171** [IndySockets/Indy#614] [Q&A]
Hi,
We had tried to update the latest IndyProtocols70 for Delphi 7 and it failed.

Can anyone advise us how to do fix it?

![Error](https://github.com/user-attachments/assets/bd60fcc8-b102-4abb-afb3-3e32b1a3d27a)

Regards,
K.H.Wong

**Comment by rlebeau** [comment]
The `UTCTimeToLocalTime()` function did not exist yet in the version of Indy that is pre-installed in Delphi 7.  Did you first remove the pre-installed version before compiling/installing a new version?

https://github.com/IndySockets/Indy/wiki/Updating-Indy

Make sure you also look for Indy binaries in Windows system folders, as old Delphi versions did not limit themselves to putting their BPLs in just Borland folders.  Try using [SysInternals Process Monitor](https://learn.microsoft.com/en-us/sysinternals/downloads/procmon) to see exactly which binaries are actually being touched during compiling/loading.

Also, the error message itself is troubling, as the `IdGlobal` unit is implemented in the `IndySystem` package, not in the `IndyProtocols` package, so nothing should be looking for the `UTCTimeToLocalTime()` function in the `IndyProtocols` package.

**Comment by khwong171** [comment]
I had tried to follow up all the steps https://github.com/IndySockets/Indy/wiki/Updating-Indy but the problem seems persistent
I will try to use sysinternals Process Monitor to do the checking... Later update.

**Comment by khwong171** [comment]
Found out the real issue. The old Indy bpl still exists... remove all the old indy*.bpl by global search.
Now it is successfully compiled.
Thanks a lot for your help.
